react/components/__tests__/QuantitySelector.test.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react'
+import { render, screen, fireEvent } from '@vtex/test-tools/react'
+import { ToastContext } from '@vtex/styleguide/lib/ToastProvider'
+
+import QuantitySelector from '../QuantitySelector'
+
+describe('<QuantitySelector />', () => {
+  it('should change from dropdown to input after quantity exceeds max value', () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector id="1" value={5} maxValue={50} onChange={onChange} />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toBeInTheDocument()
+
+    expect(dropdown).toHaveValue('5')
+
+    fireEvent.change(dropdown, { target: { value: '10' } })
+
+    expect(onChange).toHaveBeenLastCalledWith(10)
+    expect(dropdown).not.toBeInTheDocument()
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toHaveValue('10')
+  })
+
+  it('should change back to input after changing the value to under max value', () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector id="1" value={10} maxValue={50} onChange={onChange} />
+    )
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toBeInTheDocument()
+
+    fireEvent.change(input, { target: { value: '5' } })
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenLastCalledWith(5)
+    expect(input).not.toBeInTheDocument()
+  })
+
+  it("shouldn't display default measurement unit value", () => {
+    render(
+      <QuantitySelector
+        id="1"
+        value={1}
+        maxValue={50}
+        onChange={jest.fn()}
+        measurementUnit="un"
+      />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toHaveDisplayValue('1')
+  })
+
+  it("shouldn't display raw quantity value when unit multiplier exists", () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector
+        id="1"
+        value={1}
+        maxValue={50}
+        onChange={onChange}
+        unitMultiplier={0.5}
+        measurementUnit="kg"
+      />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toHaveValue('1')
+    expect(dropdown).toHaveDisplayValue('0.5 kg')
+
+    let options = screen.getAllByRole('option')
+
+    // Get the first half options, because we have two
+    // dropdowns in the page
```

Comment:
how come there are two dropdowns?

Reply:
one for mobile and the other for desktop 😆

this should be the default behavior in styleguide, but we did this because we need the size of the dropdown to be "large" in mobile and "normal" when not in mobile, and we didn't wanted to use javascript to do something we should be doing with CSS

Reply:
wow, interesting. But keeping both dropdowns does create an a11y problem though, right?

Reply:
it does not, because at least one of them is hidden using `display: none` when the other is being shown, so the screen reader never announces the hidden dropdown

Reply:
nice, thanks for the clarification